products/views.py

```diff
@@ -1,3 +1,62 @@
-from django.shortcuts import render
+import datetime
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+
+from dr_martens.checkitem   import CheckItem
+from dr_martens.login_check import login_check
+from products.models        import (Product,
+                                    ProductImage,
+                                    ProductOption)
+
+class DetailView(View):
+    @login_check
+    def get(self, request, *args, **kwargs):
+        try:
+            product           = Product.objects.get(id=kwargs["product_id"])
+            product_name      = product.korean_name
+            product_id        = product.id
+            material          = product.material
+            country_of_origin = product.country_of_origin
+
+            product_images    = ProductImage.objects.filter(product=product)
+            product_options   = ProductOption.objects.filter(product=product)
+            
+            price = product_options[0].price
+    
+            colors = []
+            sizes_stocks = []
+            for product_option in product_options:
+                colors.append(product_option.color.color)
+                sizes_stocks.append(
+                    {
+                        "size":product_option.size.name,
+                        "quantity":product_option.stock
+                    }        
+                )
+
+            colors = list(set(colors))
+             
+            img_urls = []
+            for product_image in product_images:
+                img_urls.append(product_image.url)
+
+            result = {
+                "result":{
+                    "id":product_id,
+                    "productName": product_name,
+                    "imageUrls": img_urls,
+                    "country": country_of_origin,
+                    "material": material,
+                    "centerSize": sizes_stocks,
+                    "centerColor": colors,
+                    "price": price,
+                }
+            }
```

Comment:
```suggestion
            result = {
                    "id":product_id,
                    "productName": product_name,
                    "imageUrls": img_urls,
                    "country": country_of_origin,
                    "material": material,
                    "centerSize": sizes_stocks,
                    "centerColor": colors,
                    "price": price,
            }
```
result로 한 번더 감싼 이유가 있을까요? 없다면 빼도 될 것 같은데 어떻게 생각 하시나요?

Reply:
사실 JSON 포맷에 message:success를 넣고 싶었고, 따라서 result를 이중 딕셔너리 형태로 만들어야 했었습니다.
생각해보니 status code를 확인하면 success라는 메세지 없이도 통신이 성공했다는 것을 알 수 있을 것 같습니다.
따라서 의견 반영하여 result 구조를 변경하였습니다.